components/layout_2020/flow/inline.rs

```diff
@@ -982,80 +992,45 @@ impl InlineFormattingContext {
 impl InlineBoxContainerState {
     fn new(
         inline_box: &InlineBox,
-        inline_position: &mut Length,
         containing_block: &ContainingBlock,
         text_decoration_of_parent: TextDecorationLine,
         nested_block_size_of_parent: Length,
         layout_context: &LayoutContext,
+        initial_index_in_line_items: usize,
+        is_last_fragment: bool,
     ) -> Self {
         let style = inline_box.style.clone();
-        let mut pbm = style.padding_border_margin(containing_block);
-
-        if inline_box.first_fragment {
-            *inline_position += pbm.padding.inline_start +
-                pbm.border.inline_start +
-                pbm.margin.inline_start.auto_is(Length::zero);
-        } else {
-            pbm.padding.inline_start = Length::zero();
-            pbm.border.inline_start = Length::zero();
-            pbm.margin.inline_start = LengthOrAuto::zero();
-        }
-
         let text_decoration_line = text_decoration_of_parent | style.clone_text_decoration_line();
         Self {
             base: InlineContainerState {
-                line_items_so_far: Vec::with_capacity(inline_box.children.len()),
                 has_content: false,
                 text_decoration_line,
                 nested_block_size: nested_block_size_of_parent
                     .max(line_height_from_style(layout_context, &style)),
             },
             style,
             base_fragment_info: inline_box.base_fragment_info,
-            pbm,
-            was_part_of_previous_line: false,
+            pbm: inline_box.style.padding_border_margin(containing_block),
+            line_item_index: initial_index_in_line_items,
+            is_last_fragment,
         }
     }
 
     fn layout_into_line_item(
         &mut self,
         layout_context: &LayoutContext,
-        inline_position: &mut Length,
-        at_end_of_inline_element: bool,
+        is_first_fragment: bool,
     ) -> InlineBoxLineItem {
         // If we are finishing in order to fragment this InlineBox into multiple lines, do
```

Comment:
Is this comment still relevant, now that we've removed `at_end_of_inline_element`?